src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -238,26 +235,17 @@ public TypeDesc GetTypeOfRuntimeType()
 
         protected override void ComputeDependencyNodeDependencies(List<DependencyNodeCore<NodeFactory>> obj)
         {
-            foreach (DependencyNodeCore<NodeFactory> dependency in obj)
+            PerfEventSource.Log.JitSectionStart();
+            ConditionalWeakTable<Thread, CorInfoImpl> cwt = new ConditionalWeakTable<Thread, CorInfoImpl>();
+            Parallel.ForEach(obj, dependency =>
```

Comment:
>Parallel.ForEach [](start = 12, length = 16)

With parallelism, debugging sometimes becomes a bit harder, and having a way to make things single threaded is very useful. You should add some logic here that reads and environment variable, and if it is set, execute the delegate sequentially in a simple loop outside of the Parallel.ForEach call (not sure if you can control Parallel.ForEach to run single threaded... maybe you can)